lnd.go

```diff
@@ -651,28 +697,125 @@ func Main(cfg *Config, lisCfg ListenerCfg, implCfg *ImplementationCfg,
 
 // getTLSConfig returns a TLS configuration for the gRPC server and credentials
 // and a proxy destination for the REST reverse proxy.
-func getTLSConfig(cfg *Config) ([]grpc.ServerOption, []grpc.DialOption,
-	func(net.Addr) (net.Listener, error), func(), error) {
+func getTLSConfig(cfg *Config, keyRing keychain.KeyRing) (
```

Comment:
Ugh... This function is in desperate need for refactoring. Should we create its own `tlsConfig` struct that is returned here? That struct could also support reloading and so on and have nice, clean, testable methods.